libsolidity/parsing/Parser.cpp

```diff
@@ -1552,7 +1552,13 @@ ASTPointer<Expression> Parser::parseLeftHandSideExpression(
 		{
 			m_scanner->next();
 			nodeFactory.markEndPosition();
-			expression = nodeFactory.createNode<MemberAccess>(expression, expectIdentifierToken());
+			if (m_scanner->currentToken() == Token::Address)
```

Comment:
I wonder whether this could create any problems, but I don't think so.

Reply:
I did add the address member access to structs as test because of this - I think it should be fine and just generate a member not found message for other types.